distributed/scheduler.py

```diff
@@ -4368,23 +4368,40 @@ def update_graph(
         if isinstance(annotations, ToPickle):  # type: ignore
             # FIXME: what the heck?
             annotations = annotations.data  # type: ignore
+
         stimulus_id = stimulus_id or f"update-graph-{time()}"
         (
             dsk,
             dependencies,
-            layer_annotations,
-            pre_stringified_keys,
-        ) = self.materialize_graph(graph)
```

Comment:
previously I passed throuhg non-stringified keys to have them available when dealing with annotations at a later point. it's much better to get this out of the way and do this here.
Once/if stringification is removed this should simplify things with annotations a lot